src/DFTK.jl

```diff
@@ -112,12 +114,15 @@ export scf_anderson_solver
 export scf_CROP_solver
 export self_consistent_field
 export direct_minimization
+export newton
+export newton_step
```

Comment:
Feels internal, why export?

Reply:
I know in advance that I will need to use newton_step to test stuff we are working on with Antoine, Éric and Geneviève. But if you prefer we just export newton :)

Reply:
I'd say that's a niche use case. You can always explicitly use it still, so I'd not export it.